packages/grafana-ui/src/components/Graph/utils.ts

```diff
@@ -119,19 +119,45 @@ export const graphTimeFormat = (ticks: number | null, min: number | null, max: n
     const oneYear = 31536000000;
 
     if (secPerTick <= 45) {
-      return 'HH:mm:ss';
+      return localTimeFormat({ hour: '2-digit', minute: '2-digit', second: '2-digit', hour12: false }, 'HH:mm:ss');
     }
     if (secPerTick <= 7200 || range <= oneDay) {
-      return 'HH:mm';
+      return localTimeFormat({ hour: '2-digit', minute: '2-digit', hour12: false }, 'HH:mm');
     }
     if (secPerTick <= 80000) {
-      return 'MM/DD HH:mm';
+      return localTimeFormat(
+        { month: '2-digit', day: '2-digit', hour: '2-digit', minute: '2-digit', hour12: false },
+        'MM/DD HH:mm'
+      );
     }
     if (secPerTick <= 2419200 || range <= oneYear) {
-      return 'MM/DD';
+      return localTimeFormat({ month: '2-digit', day: '2-digit', hour12: false }, 'MM/DD');
     }
-    return 'YYYY-MM';
+    return localTimeFormat({ year: 'numeric', month: '2-digit', hour12: false }, 'YYYY-MM');
   }
 
-  return 'HH:mm';
+  return localTimeFormat({ hour: '2-digit', minute: '2-digit', hour12: false }, 'HH:mm');
+};
+
+const localTimeFormat = (options: Intl.DateTimeFormatOptions, fallback: string): string => {
+  if (!window.Intl) {
+    return fallback;
+  }
+
+  // https://momentjs.com/docs/#/displaying/format/
+  const parts = new Intl.DateTimeFormat('default', options).formatToParts(new Date());
+  const mapping: { [key: string]: string } = {
+    year: 'YYYY',
+    month: 'MM',
+    day: 'DD',
+    hour: 'HH',
+    minute: 'mm',
+    second: 'ss',
+    weekday: 'ddd',
+    era: 'N',
+    dayPeriod: 'A',
+    timeZoneName: 'Z',
+  };
```

Comment:
This is  called a lot of times, wonder if there is a possibility to cache/memoize this

Reply:
Yes, In fact it can be pre-generated once. I will make that change.